### Part I
For the system to be operational, it needs at least one controller and 2 actuators to be working

Let $C_{i}$ be the event that $i$ controllers are working, and let $C_j$ be the event that $j$ controllers are working. Let $p_c$ be the event that a controller fails and $p_a$ be the event that an actuator fails. Let $N_c$ and $N_a$ be the number of controllers and actuators, respectively.

Then the system will be operational $(O)$  if 
$$\overline{C_0} \cap \overline{A_0 \cup A_1}.$$

Thus,
$$
P(O) = \left[ 1-p_{c}^{N_c} \right]
\left[ 1-p_{a}^{N_a} - N_a (1-p_a)p_a ^{N_a-1} \right]
$$

In [1]:
def operational(num_c, num_a, pc=0.25, pa=0.1):
    op_c=1-pc**num_c
    op_a=1-pa**num_a-num_a*(1-pa)*pa**(num_a-1)
    return op_c*op_a
    

In [2]:
operational(1,2)

0.6074999999999999

In [3]:
operational(2,2)

0.7593749999999999

In [4]:
operational(2,3)

0.91125

In [5]:
operational(2,4)

0.93403125

In [6]:
operational(3,4)

0.9807328125

### Part II

I apologize that this problem was a little ambiguous. Let's evaluate the answer for specific values of $p_c$ at system (e). To be more clear, let's denote the probability that a controller or actuator of system (e) have failed as $p_{c,e}$. We can use the formula/function above to find $P(O_e)$, the probability that the system of part (e)is operational.  

Then we can solve for the needed value of $p_{c,a}$, the probability that a controller or actuator of system (a) has failed.  Note that for part (a), we can write
$$
P(O)= (1-p_{c,a})(1-p_{c,a})^2 = (1-p_{c,a})^3
$$
Thus, we wish to find the value of $p_{c,a}$ that satisfies
$$
p_{c,a}=1- \sqrt[3]{P(O_e)}
$$

In [7]:
operational(3,4,0.1,0.1)

0.9953037

In [8]:
1-(0.9953037)**(1/3)

0.0015678903286447943

The required probability of component failure for system (a) is almost two orders of magnitude lower than for system (e)

In [9]:
operational(3,4,0.25,0.25)

0.93438720703125

In [10]:
1-(0.9343872)**(1/3)

0.022367508577909367

Here, the required probability of component failure for system (a) is slightly more than one order of magnitude lower then for system (e)

In [11]:
operational(3,4,0.01,0.01)

0.9999950300039699

In [12]:
1-(0.9999950300039699)**(1/3)

1.6566680879348539e-06

As the components of system (e) become more reliable, the requirements on the components of system (a) become even more stringent. For system (e) operating with probability of component failure of $10^{-2}$, system (a) would need components that fail with probability $1.66 \times 10^{-6}$ to achieve the same system reliability